### PR TITLE
COMMERCE-4845 Attempting to delete a base price list or base sale price list throws an error but can be done

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/jaxrs/exception/mapper/CommerceBasePriceListCannotDeleteExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(
+	enabled = false,
+	property = {
+		"osgi.jaxrs.application.select=(osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Pricing)",
+		"osgi.jaxrs.extension=true",
+		"osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Pricing.CommerceBasePriceListCannotDeleteExceptionException"
```
Comment (dsitu) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/jaxrs/exception/mapper/CommerceBasePriceListCannotDeleteExceptionMapper.java:34]: For headless we should match the exception name with the rest name(PriceList).
```suggestion
		"osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Pricing.BasePriceListCannotDeleteExceptionException"
```

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/jaxrs/exception/mapper/CommerceBasePriceListCannotDeleteExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	},
+	service = ExceptionMapper.class
+)
+@Provider
+public class CommerceBasePriceListCannotDeleteExceptionMapper
```
Comment (dsitu) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/jaxrs/exception/mapper/CommerceBasePriceListCannotDeleteExceptionMapper.java:39]: For headless we should match the exception name with the rest name(PriceList).
```suggestion
public class BasePriceListCannotDeleteExceptionMapper
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:forward

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:forward

**Pull request closed without merging**